packages/loader/driver-utils/src/parallelRequests.ts

```diff
@@ -29,12 +31,15 @@ export class ParallelRequests<T> {
     private latestRequested: number;
     private nextToDeliver: number;
     private readonly results: Map<number, T[]> = new Map();
-    private working = true;
+    private workingState: WorkingState = "working";
```

Comment:
Should it be intialized with done state as "working" means it is in process of fetching something.

Reply:
"working" means not canceled done, i.e. there is no separation between "started to work" and "did not start to work"

Reply:
ok. So as per the code it seems that once it moves to "cancel" or "done" state, it will never come to "working" state again. "Working" can only be the initial state.

Reply:
Correct